pages/blog/20190831-newfeature.md

```diff
@@ -0,0 +1,122 @@
+# Adding a New Feature
+
+Aug 31, 2019 • [irisb1701](https://github.com/irisb1701)
+
+---
+
+As a system's engineer you may need to work with the treehouses cli (command line interface). 
+
+Here are step by step instructions on how to create a new feature for treehouses.
+
+## Determining a New Feature:
+
+Example situation:
+As you're testing the new image for treehouses you notice your raspberry pi is getting hot. You know anything past 85°C may damage your pi so you'd like to monitor its core temperature.
+
+You already know the raspberry pi command `vcgencmd measure_temp` will return its core temperature in celsius.
+
+Looking at the treehouses cli commands, you notice there aren't any commands involving detecting temperature and decide it would be a useful addition.
+
+Now that you have an idea for a new command, you will create a new issue and when approved, can create a new branch and begin working on it.
+
+
+## Step 0 - Vim
+
+Vim is an easy way to test your code. Here are instructions on how to use it.
+
+1. Download Vim  (or type `brew install vim` in your terminal)
```

Comment:
Add link to Vim - https://www.vim.org/download.php

Maybe also differentiate between Windows and macOS?